lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -35,14 +54,16 @@ def get_problem_grade_distribution(course_id):
       'total_student_count' where the key is problem 'module_id' and the value is number of students
         attempting the problem
     """
+    non_student_list = get_non_student_list(course_id)
 
     # Aggregate query on studentmodule table for grade data for all problems in course
-    db_query = models.StudentModule.objects.filter(
+    db_query = models.StudentModule.objects.select_related('student').filter(
         course_id__exact=course_id,
         grade__isnull=False,
         module_type__in=PROB_TYPE_LIST,
-    ).values('module_state_key', 'grade', 'max_grade').annotate(count_grade=Count('grade'))
+    ).exclude(student__id__in=non_student_list).values('module_state_key', 'grade', 'max_grade').annotate(count_grade=Count('grade'))
```

Comment:
Because we expect `non_student_list` to be small, passing a list into the queryset "in" parameter is the right thing to do .  In degenerate cases where the list might be huge (>100K elems), this might not be the right thing.